src/sentry/snuba/tasks.py

```diff
@@ -116,31 +112,28 @@ def delete_subscription_from_snuba(query_subscription_id):
 
 
 def _create_in_snuba(subscription):
-    conditions = resolve_discover_aliases(get_filter(subscription.query))[0].conditions
-    try:
-        environment = subscription.environments.all()[:1].get()
-    except Environment.DoesNotExist:
-        environment = None
-
-    if environment:
-        conditions.append(["environment", "=", environment.name])
-    conditions = apply_dataset_conditions(QueryDatasets(subscription.dataset), conditions)
+    snuba_query = subscription.snuba_query
+    snuba_filter = get_filter(snuba_query.query)
+    snuba_filter.update_with(
+        resolve_field_list([snuba_query.aggregate], snuba_filter, auto_fields=False)
+    )
+    snuba_filter = resolve_discover_aliases(snuba_filter)[0]
```

Comment:
@markstory This looks like it works correctly, just wanted your confirmation. We're using the function format, so `count()`, `count_unique(tags[sentry:user])`, etc.